compiler/rustc_const_eval/src/interpret/place.rs

```diff
@@ -793,6 +793,31 @@ where
         }
     }
 
+    pub fn write_uninit(&mut self, dest: &PlaceTy<'tcx, M::PointerTag>) -> InterpResult<'tcx> {
+        let mplace = match dest.place {
+            Place::Ptr(m) => m,
+            Place::Local { frame, local } => {
+                match M::access_local_mut(self, frame, local)? {
+                    Ok(local) => {
+                        *local = LocalValue::Uninitialized;
```

Comment:
I'm afraid that is the wrong kind of "uninit". `LocalValue::Uninitialized` is a hack that IMO shouldn't exist but is currently required to support unsized local variables.

This is wrong because it leaks memory (if the local was live), and because it will mean that the local might actually *change its address*, which doesn't make a lot of sense.

Instead this should create a suitable uninit `Immediate` and then set the local do `LocalValue::Live(Operand::Immediate(uninit))`.

Reply:
Ah, that note about using `layout.abi` was super helpful. I've updated this, hopefully it's now doing the right thing